libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -249,69 +244,75 @@ proc getGossipPeers(g: GossipSub): Table[string, ControlMessage] {.gcsafe.} =
       result[id].ihave.add(ihave)
 
 proc heartbeat(g: GossipSub) {.async.} =
-  while g.heartbeatRunning:
-    try:
-      trace "running heartbeat"
+  # don't lock on every iteration
+  withLock g.heartbeatLock:
```

Comment:
should the sleep be locked??

Reply:
You don't want two heartbeats running in concurrently, this is mainly to prevent that.

Reply:
Sleep should NOT be locked, I think it's not tho